base_geoengine/models/base.py

```diff
@@ -0,0 +1,169 @@
+            "default_extent": view.default_extent or DEFAULT_EXTENT,
+            "default_zoom": view.default_zoom,
+        }
+
+        for layer in view.raster_layer_ids:
+            layer_dict = layer.read()[0]
+            geoengine_layers["backgrounds"].append(layer_dict)
+        for layer in view.vector_layer_ids:
+            layer_dict = layer.read()[0]
+            layer_dict["attribute_field_id"] = set_field_real_name(
+                layer_dict.get("attribute_field_id", False)
+            )
+            layer_dict["geo_field_id"] = set_field_real_name(
+                layer_dict.get("geo_field_id", False)
+            )
+            layer_dict["model"] = layer.model_id.model
+            layer_dict["model_domain"] = layer.model_domain
+            geoengine_layers["actives"].append(layer_dict)
+        return geoengine_layers
+
+    @api.model
+    def get_edit_info_for_geo_column(self, column):
+        raster_obj = self.env["geoengine.raster.layer"]
+
+        field = self._fields.get(column)
+        if not field or not isinstance(field, geo_fields.GeoField):
+            raise ValueError(
+                _("%s column does not exists or is not a geo field") % column
+            )
+        view = self._get_geo_view()
+        raster = raster_obj.search(
+            [("view_id", "=", view.id), ("use_to_edit", "=", True)], limit=1
+        )
+        if not raster:
+            raster = raster_obj.search([("view_id", "=", view.id)], limit=1)
+        if not raster:
+            raise MissingError(_("No raster layer for view %s") % (view.name,))
+        return {
+            "edit_raster": raster.read()[0],
+            "srid": field.srid,
+            "projection": view.projection,
+            "restricted_extent": view.restricted_extent,
+            "default_extent": view.default_extent or DEFAULT_EXTENT,
+            "default_zoom": view.default_zoom,
+        }
+
+    @api.model
+    def geo_search(
+        self, domain=None, geo_domain=None, offset=0, limit=None, order=None
+    ):
+        """Perform a geo search it allows direct domain:
+        geo_search(
+            domain=[('name', 'ilike', 'toto']),
+            geo_domain=[('the_point', 'geo_intersect',
+                          myshaply_obj or mywkt or mygeojson)])
+
+        We can also support indirect geo_domain (
+           ‘geom’, ‘geo_operator’, {‘res.zip.poly’: [‘id’, ‘in’, [1,2,3]] })
+
+        The supported operators are :
+         * geo_greater
+         * geo_lesser
+         * geo_equal
+         * geo_touch
+         * geo_within
+         * geo_contains
+         * geo_intersect"""
+        # First we do a standard search in order to apply security rules
+        # and do a search on standard attributes
+        # Limit and offset are managed after, we may loose a lot of performance
+        # here
+        _logger.debug(
+            "geo_search is deprecated: uses search method defined on base model"
+        )
```

Comment:
@kouffsamuel We can use here the standard way to declare deprecation into python....
 In your imports add: `import warnings`

```suggestion
        warnings.warn(
           "geo_search is deprecated: uses search method defined on base model",
        DeprecationWarning)
```